src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -366,6 +375,18 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
     }
   }
 
+  /**
+   * Handles the key up event with MatSelect.
+   * Allows e.g. the announcing of the currently activeDescendant by screen readers.
+   * @param event
+   */
+  _handleKeyup(event: KeyboardEvent) {
+    const activeDescendant = this._options.find((item) => item.id === this.matSelect._getAriaActiveDescendant());
+    const index = this._options.toArray().findIndex(item => item === activeDescendant);
+    this.liveAnnouncer.announce(
+      ' ' + activeDescendant.viewValue + index + this.indexAndLengthScreenReaderText + (this._options.toArray().length - 1)
```

Comment:
```suggestion
      ' ' + activeDescendant.viewValue + ' ' + (index + 1) + this.indexAndLengthScreenReaderText + this._options.toArray().length
```

Reply:
Heya , 

this one is a bit odd, where the search input counts as an option so it works best the other way around. 

kinda looks like [ Searchbox, 1, 2, 3, 4] so when you are selecting option 1 is reads 1 of (5-1) which makes sense to users. 

Thoughts?

Reply:
right, didn't think of that.
you could extract parts of the following code into a separate method and use it to determine the length offset:

https://github.com/bithost-gmbh/ngx-mat-select-search/blob/d19c777ded98457cbdd4be476246ae96c52d7f02/src/app/mat-select-search/mat-select-search.component.ts#L586-L590

Reply:
Sorry i don't understand what you mean. this._options.lenght can never be 0 as it always contains the search input and it was my understanding that the length offset is always 1 because of that.

Reply:
previously it was possible to add the search input without wrapping it in a mat-option, which is why there has to be this distinction, see also https://github.com/bithost-gmbh/ngx-mat-select-search/blame/master/README.md#L64-L65